kong/plugins/acl/handler.lua

```diff
@@ -1,47 +1,76 @@
-local singletons = require "kong.singletons"
-local BasePlugin = require "kong.plugins.base_plugin"
-local cache = require "kong.tools.database_cache"
-local responses = require "kong.tools.responses"
-local utils = require "kong.tools.utils"
-local constants = require "kong.constants"
+local singletons   = require "kong.singletons"
+local BasePlugin   = require "kong.plugins.base_plugin"
+local cache        = require "kong.tools.database_cache"
+local responses    = require "kong.tools.responses"
+local utils        = require "kong.tools.utils"
+local constants    = require "kong.constants"
+
 
 local table_insert = table.insert
 local table_concat = table.concat
-local ipairs = ipairs
-local empty = {}
+local ngx          = ngx
+local set_header   = ngx.req.set_header
+local ipairs       = ipairs
+local type         = type
+local next         = next
+local empty        = {}
+
 
 local ACLHandler = BasePlugin:extend()
 
+
 ACLHandler.PRIORITY = 950
 
+
 function ACLHandler:new()
   ACLHandler.super.new(self, "acl")
 end
 
+
 local function load_acls_into_memory(consumer_id)
-  local results, err = singletons.dao.acls:find_all {consumer_id = consumer_id}
+  local results, err = singletons.dao.acls:find_all { consumer_id = consumer_id }
   if err then
     return nil, err
   end
   return results
 end
 
+
 function ACLHandler:access(conf)
   ACLHandler.super.access(self)
 
   local consumer_id
-  if ngx.ctx.authenticated_credential then
-    consumer_id = ngx.ctx.authenticated_credential.consumer_id
-  else
-    return responses.send_HTTP_FORBIDDEN("Cannot identify the consumer, add an authentication plugin to use the ACL plugin")
+  local ctx = ngx.ctx
+
+  local authenticated_consumer = ctx.authenticated_consumer
+  if authenticated_consumer and type(authenticated_consumer) == "table" then
+    consumer_id = authenticated_consumer.id
+  end
+
+  if not consumer_id then
+    local authenticated_credential = ctx.authenticated_credential
+    if authenticated_credential and type(authenticated_credential) == "table" then
+      consumer_id = authenticated_credential.consumer_id
+    end
+  end
+
+  if not consumer_id then
+    return responses.send_HTTP_FORBIDDEN(
+      "Cannot identify the consumer, add an authentication plugin to use the ACL plugin"
+    )
   end
 
   -- Retrieve ACL
-  local acls, err = cache.get_or_set(cache.acls_key(consumer_id), nil,
-                                load_acls_into_memory, consumer_id)
+  local acls, err = cache.get_or_set(
+    cache.acls_key(consumer_id),
+    nil,
+    load_acls_into_memory,
+    consumer_id)
```

Comment:
Any reason for this change and the new line below? I think it was fine as it was (per the code style), except for the second line not aligning with the first argument. Such style changes should be part of another commit/PR anyways.